### ISSUE TITLE
missing snap features

### ISSUE THREAD

**Issue by xyont** [Macad3D/Macad3D#61]
i'm not really know how hard it is to be implemented, but the snap feature of vertex in a part is an important tool for precision modeling.

**Comment by dipts** [comment]
Snapping to vertices is basically supported and can be enabled in parallel to the other snap modes. But unfortunately not every tool supports snapping per se yet. 

In which specific tool/workflow did you miss snapping?

**Comment by dipts** [comment]
[Version 4.0](https://macad3d.net/userguide/go/?guid=0aa98281-d673-4261-af37-cee77ada8add) contains an extensively refactorised and expanded snapping system. If you still have any requests, please open a new ticket with a description of the desired behaviour.

**Comment by xyont** [comment]
thank you so much for improvement, however base point as reference in moving part still missing.
    On Thursday, January 2, 2025 at 04:42:29 AM GMT+7, dipts ***@***.***> wrote:  
 
 


Version 4.0 contains an extensively refactorised and expanded snapping system. If you still have any requests, please open a new ticket with a description of the desired behaviour.

**Comment by dipts** [comment]
Thanks for your feedback. Can you please create a new issue for this new feature and describe in more detail what you have in mind? Perhaps with a drawing or screenshot to help us understand.
